smart-farm-backend/src/main/java/api/smartfarm/services/SectorService.java

```diff
@@ -46,4 +50,25 @@ public List<SectorCropTypesDTO> getSectorsCropTypes(String farmId) {
             return new SectorCropTypesDTO(sector, cropType);
         }).collect(Collectors.toList());
     }
+
+    public SectorDTO create(String farmId, SectorDTO sectorDTO) {
+        Farm farm = farmService.getFarmById(farmId);
+        LOGGER.info("Getting farmId: {} for create sectors", farmId);
+        farm.getSectors().add(new Sector(sectorDTO));
+        farmDAO.save(farm);
+        LOGGER.info("Saved farm {} successfully", farm);
+        return sectorDTO;
```

Comment:
Cambiaría el método para que quede así:

`Farm farm = farmService.getFarmById(farmId);
 LOGGER.info("Adding sector {} to farm {}", sectorDTO.getCode(), farmId);
 farm.getSectors().add(new Sector(sectorDTO));
 farmService.update(farm);
 return sectorDTO;`

Borrando farmDAO de esta clase